### ISSUE TITLE
Upstream kernel update workflow

### ISSUE THREAD

**Issue by ollieparanoid** [postmarketOS/pmbootstrap#1237]
### Status Quo
We have three upstream kernels (directly from [kernel.org](https://kernel.org)) packaged:
* `linux-postmarketos-mainline` (most bleeding edge, we allow device specific patches on top as long as they are getting upstreamed in parallel)
* `linux-postmarketos-stable`
* `linux-postmarketos-lts`

Right now the update policy is, that whoever wants to update it, makes a PR (@craftyguy did it for a time, thanks a lot!). The reality is, that all upstream kernel packages are outdated by now.

### Proposal
I would like to make updating the kernels as easy as possible. Here's what the user updating the kernels would do:
* Add `pmbootstrap aportgen linux-postmarketos-lts` (and all other variants):
  * Looks up the current kernel versions from `kernel.org`
  * Generates the kernel APKBUILDs (which are long and all have duplicated code right now, so this makes it easier to maintain in general)
  * Runs `pmbootstrap checksum` internally to download the kernel tarballs and create the checksums
* Try to build all updated kernels (`pmbootstrap build linux-postmarketos-mainline linux-postmarketos-stable linux-postmarketos-lts`)
* When a device specific patch we're carrying fails to apply in the mainline packaging:
  * **Just throw that patch out.** Either it has been upstreamed already, or it is incompatible now.
  * Note in the device's wiki page which commit was the last one that had the patch included.
* When all kernels build, commit it directly to the `pmbootstrap` `master` branch (if the user has write access, that's currently @MartijnBraam and me) or make a PR for the update.

### Maintainer
I'd be happy if someone jumped in and said they would like to do it. If no one shows up, I would do it.

What do you think, folks?

**Comment by z3ntu** [comment]
I could do it. :) 1. That makes me put more time into pmb ^^ and 2. It's not too much work.

**Comment by craftyguy** [comment]
There should probably be some lightweight smoke testing in the process too. (e.g. boot on qemu + 1 physical device)? I recall there being a couple of instances when I was pushing updates where the config for the new kernel broke something (qemu).

**Comment by ollieparanoid** [comment]
Good idea! For Qemu we already have an automated test case nowadays: `test/test_qemu_running_processes.py`. That combined with testing on one device should do it I think.
